### PR TITLE
remove unused definitions, params, args from sandbox Scala code

### PR DESCRIPTION
**Pull request by S11001001**
As found by `-Ywarn-unused` in #6907.

* [x] requires #6992 merge for compilation

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala [hunk 1]
```diff
@@ -360,13 +360,12 @@ private[indexer] class JdbcIndexer private[indexer] (
...
-                    .future(
-                      metrics.daml.indexer.stateUpdateProcessing,
-                      handleStateUpdate(offset, update),
-                    )
+              withEnrichedLoggingContext(JdbcIndexer.contextFor(offset, update)) { _ =>
+                Timed
+                  .future(
+                    metrics.daml.indexer.stateUpdateProcessing,
+                    handleStateUpdate(offset, update),
+                  )
```
Comment (S11001001) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:368]: Same considerations in play as in https://github.com/digital-asset/daml/pull/6992/files#r465275865

Comment (SamirTalwar-DA) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:368]: This looks like a bug. The logging context should be passed through to `handleStateUpdate`. Instead it's using one that's passed in at the class constructor level. I'll open a separate fix.

Comment (SamirTalwar-DA) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:368]: Fixed in #7042. Please take a look when you can.

Comment (S11001001) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala]: Included in this PR's build.

**Pull request merged**